core/test/base/range_accessors.cpp

```diff
@@ -121,4 +142,901 @@ TEST_F(RowMajorAccessor, CanAssignSubranges)
+                                                0b0101>;
+
+    using reduced_storage = gko::range<accessor>;
+    using const_reduced_storage = gko::range<const_accessor>;
+
+    const gko::dim<3> size{1u, 4u, 2u};
+    static constexpr gko::size_type data_elements{8};
+    static constexpr gko::size_type scalar_elements{8};
+    // clang-format off
+    st_type data[8]{
+        10, 11,
+        -12, 13,
+        14, -115,
+        6, 77
+    };
+    ar_type scalar[scalar_elements]{
+        1., 2., 3., 4., 5., 6., 7., 8.
+    };
+    // clang-format on
+    const std::array<gko::size_type, 2> storage_stride{{8, 2}};
+    const std::array<gko::size_type, 1> scalar_stride{{2}};
+    reduced_storage r{size, data, storage_stride, scalar, scalar_stride};
+    const_reduced_storage cr{size, data, scalar};
+
+    template <typename Accessor>
+    static void check_accessor_correctness(
+        const Accessor &a,
+        std::tuple<int, int, int> ignore = std::tuple<int, int, int>(99, 99,
+                                                                     99))
+    {
+        // Test for equality is fine here since they should not be modified
+        // clang-format off
+        if (ignore != t(0, 0, 0)) { EXPECT_EQ(a(0, 0, 0), ar_type{10.});   }
+        if (ignore != t(0, 0, 1)) { EXPECT_EQ(a(0, 0, 1), ar_type{22.});   }
+        if (ignore != t(0, 1, 0)) { EXPECT_EQ(a(0, 1, 0), ar_type{-12.});  }
+        if (ignore != t(0, 1, 1)) { EXPECT_EQ(a(0, 1, 1), ar_type{26.});   }
+        if (ignore != t(0, 2, 0)) { EXPECT_EQ(a(0, 2, 0), ar_type{14.});   }
+        if (ignore != t(0, 2, 1)) { EXPECT_EQ(a(0, 2, 1), ar_type{-230.}); }
+        if (ignore != t(0, 3, 0)) { EXPECT_EQ(a(0, 3, 0), ar_type{6.});    }
+        if (ignore != t(0, 3, 1)) { EXPECT_EQ(a(0, 3, 1), ar_type{154.});  }
+        // clang-format on
+    }
+};
+
+using ScaledReducedStorage3dTypes = ::testing::Types<
+    std::tuple<double, gko::int64>, std::tuple<double, gko::int32>,
+    std::tuple<double, gko::int16>, std::tuple<float, gko::int32>,
+    std::tuple<float, gko::int16>>;
+
+TYPED_TEST_SUITE(ScaledReducedStorage3d, ScaledReducedStorage3dTypes);
+
+
+TYPED_TEST(ScaledReducedStorage3d, CorrectLengths)
+{
+    EXPECT_EQ(this->r.length(0), this->size[0]);
+    EXPECT_EQ(this->r.length(1), this->size[1]);
+    EXPECT_EQ(this->r.length(2), this->size[2]);
+    EXPECT_EQ(this->r.length(3), 1);
+    EXPECT_EQ(this->r->get_size(), this->size);
+}
+
+
+TYPED_TEST(ScaledReducedStorage3d, CorrectStride)
+{
+    EXPECT_EQ(this->r->get_storage_stride()[0], this->size[1] * this->size[2]);
+    EXPECT_EQ(this->r->get_storage_stride().at(0),
+              this->size[1] * this->size[2]);
+    EXPECT_EQ(this->r->get_storage_stride()[1], this->size[2]);
+    EXPECT_EQ(this->r->get_storage_stride().at(1), this->size[2]);
+    EXPECT_EQ(this->r->get_scalar_stride(), this->scalar_stride);
+    EXPECT_EQ(this->r->get_scalar_stride(), this->scalar_stride);
```

Comment:
duplicate

Reply:
I think I was about to simplify it by just checking both arrays for equality and did not finish it for some reason...
Good find.